webextensions/background/module/metrics.js

```diff
@@ -0,0 +1,5 @@
+// Defined in a classic script source, and we can read these as global variables. 
+/* global
+  MetricsData: false,
+ */
+export const gMetricsData = new MetricsData();
```

Comment:
Why don't you separate this change as another PR? This PR looks doing too much things. Focusing to one topic for each PR, please.

Reply:
I create https://github.com/piroor/treestyletab/pull/1927/commits/5c737d8eeedd41b49624d12f41e873d5f03c1296 as minimal PoC change to convert es module.

> Focusing to one topic for each PR, please.

Okay. I'll create a new pull request.